### PR TITLE
[docs][python] add conda-forge install instructions

### PR DESCRIPTION
**Pull request by raybellwaves**
closes https://github.com/microsoft/LightGBM/issues/3542

### CODE REVIEW

File: docs/Python-Intro.rst
Comment (jameslamb) [inline docs/Python-Intro.rst:29]:
```suggestion
**Note**: The `lightgbm conda-forge package <https://github.com/conda-forge/lightgbm-feedstock>`_ is not maintained by LightGBM maintainers.
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks!

Just a small suggestion. I'll let other reviewers comment on whether we want to document this here before I approve, but in my opinion this is a good addition.

**Comment by StrikerRUS** [comment]
Why [conda-forge](https://anaconda.org/conda-forge/lightgbm) but not [official conda](https://anaconda.org/anaconda/lightgbm) channel?

File: docs/Python-Intro.rst
Comment (StrikerRUS) [reply docs/Python-Intro.rst:29]: I'm doubt that we should document conda installation at all, because we do not maintain it and hence cannot guarantee the quality and frequency of version updates. We don't aim to document all available ways to install LightGBM via all available package managers in the world.

Comment (jameslamb) [reply docs/Python-Intro.rst:29]: I actually didn't even know about https://anaconda.org/anaconda/lightgbm that you just mentioned in https://github.com/microsoft/LightGBM/pull/3544#issuecomment-723966287 . I think that's reason enough to give some guidance in our docs about how to install with `conda`.

> We don't aim to document all available ways to install LightGBM via all available package managers in the world.

Ok but this isn't an obscure package manager like `nix` or `snap`. I think it's common, in machine learning projects in Python, to prefer `conda install` to `pip install` and I think that explicitly stating "yes this exists but we don't maintain it" in our documentation is useful.

**Comment by raybellwaves** [comment]
> Why conda-forge but not official conda channel?

In my experience conda-forge is more up to date than the anaconda channel.

e.g. conda-forge is at v3.0.0 ([last update 8 days ago](https://anaconda.org/conda-forge/lightgbm/files)) and anaconda is at v2.3.0 ([last update 8 months and 11 days ago](https://anaconda.org/anaconda/lightgbm/files))

**Comment by raybellwaves** [comment]
I apologize for having some of the convo regarding this in the issue chat [here ](https://github.com/microsoft/LightGBM/issues/3542#issuecomment-724068363)

The other thing to discuss is where to put this. I feel it should go in https://github.com/microsoft/LightGBM/blob/master/python-package/README.rst#installation and the install instructions in https://lightgbm.readthedocs.io/en/latest/Python-Intro.html#install are updated from

```
You can install LightGBM via pip

pip install lightgbm
Refer to Python-package folder for the detailed installation guide.

To verify your installation, try to import lightgbm in Python:

import lightgbm as lgb
```

to

```
Refer to Python-package folder for the installation guide.

To verify your installation, try to import lightgbm in Python:

import lightgbm as lgb
```

File: docs/Python-Intro.rst
Comment (StrikerRUS) [reply docs/Python-Intro.rst]: Sorry, I'm not 100% sold on this.

Comment (jameslamb) [reply docs/Python-Intro.rst]: ok, I'll defer to what you decide.

Comment (StrikerRUS) [reply docs/Python-Intro.rst]: I think we should ask other maintainers.

Comment (jameslamb) [reply docs/Python-Intro.rst]: ok. @henry0312 or @guolinke what do you think? The question is:

> should the LightGBM Python documentation explicitly mention installing with `conda install`, even though we don't maintain the conda packages?

Comment (jameslamb) [reply docs/Python-Intro.rst]: gently ping @henry0312 or @guolinke for an opinion

Comment (guolinke) [reply docs/Python-Intro.rst]: sorry for missing this.
I think pip is the "lighter" way, as it requires fewer dependencies. So we prefer to install via pip.
Many machine learning tools have heavy dependencies, such as cuda, so they prefer conda, which pre-installs these dependencies.

However, I think we can still provide the conda install command, but clarify that we don't maintain it.

Comment (jameslamb) [reply docs/Python-Intro.rst]: thanks! Ok I agree with that. I can review from here.

File: docs/Python-Intro.rst
Comment (jameslamb) [inline docs/Python-Intro.rst:17]: @raybellwaves please revert this. We still want to make it clear that `pip install` is the preferred installation method, since we maintain the packages on PyPi.

I'd support a change where `You can install LightGBM via pip` becomes `The preferred way to install LightGBM is via pip`.

File: python-package/README.rst
Comment (jameslamb) [inline python-package/README.rst:144]:
```suggestion
Install from `conda-forge <https://anaconda.org/conda-forge/lightgbm>`_
```

File: python-package/README.rst
Comment (jameslamb) [inline python-package/README.rst:153]:
```suggestion
If you use ``conda`` to manage Python dependencies, you can install ``lightgbm`` using ``conda install``.

**Note**: The `lightgbm conda-forge package <https://github.com/conda-forge/lightgbm-feedstock>`_.  is not maintained by LightGBM maintainers.

.. code:: sh

    conda install -c conda-forge lightgbm
```

Can you please use this rewording?

**Review by jameslamb — CHANGES_REQUESTED** [review]
@raybellwaves sorry for our delay on reviewing this! Thanks for working on it. Can you please see my recent review comments?

File: python-package/README.rst
Comment (StrikerRUS) [inline python-package/README.rst:145]:
```suggestion
Install from `conda-forge channel <https://anaconda.org/conda-forge/lightgbm>`_ Using ``conda``
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
```

File: python-package/README.rst
Comment (StrikerRUS) [inline python-package/README.rst:149]:
```suggestion
**Note**: The `lightgbm conda-forge feedstock <https://github.com/conda-forge/lightgbm-feedstock>`_ is not maintained by LightGBM maintainers.
```

File: python-package/README.rst
Comment (StrikerRUS) [inline python-package/README.rst:147]: For the consistency with the rest of this document.
```suggestion
If you use ``conda`` to manage Python dependencies, you can install LightGBM using ``conda install``.
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
@raybellwaves Thanks for your contribution! I have just few minor comments.

File: python-package/README.rst
Comment (jameslamb) [reply python-package/README.rst:145]: @StrikerRUS I previously recommended removing the "Using" part (https://github.com/microsoft/LightGBM/pull/3544#discussion_r554287324). I think it makes the title more verbose but doesn't provide any additional information

Comment (StrikerRUS) [reply python-package/README.rst:145]: Ah, I see, OK!

Then please ignore my previous suggestion. And I think it will make sense to remove the "Using" part from `Install from PyPI Using pip` heading above.

Comment (raybellwaves) [reply python-package/README.rst:145]: Thanks for the discussion. I'm unsure on what the suggestion is here. You are welcome to push to my branch.

Comment (jameslamb) [reply python-package/README.rst:144]: sorry for the confusion. I just pushed https://github.com/microsoft/LightGBM/pull/3544/commits/5a91ca1a7df340c383f0d588cefb451c7b17f0a7 to resolve this thread.

Comment (jameslamb) [reply python-package/README.rst]: and https://github.com/microsoft/LightGBM/pull/3544/commits/ffd78bdb3a36a839215abbfb6d19e8f48876e0dd to address @StrikerRUS 's suggestion.

**Review by jameslamb — APPROVED** [review]
thanks Ray, looks good to me!

**Review by StrikerRUS — APPROVED** [review]
@raybellwaves Thanks a lot for your contribution!
@jameslamb Thanks for the help!

**Pull request merged**
